Calculator.Generic.INumber/RomanNumericsNumber.cs

```diff
@@ -63,33 +65,42 @@ public override string ToString()
 
         public static RomanNumericsNumber Parse(string s, IFormatProvider? provider)
         {
-            string strToParse = s.ToUpper();
+            var strToParse = s.ToUpper();
             if (string.IsNullOrEmpty(s))
             {
                 return new RomanNumericsNumber(0);
             }
-            int resultArabicNumber = 0;
-            int count = 0;
-            int len = strToParse.Length;
-            for (int i = 0; i <= len - 1; i++)
+            var resultArabicNumber = 0;
+            var count = 0;
+            var len = strToParse.Length;
+            for (var i = 0; i <= len - 1; i++)
             {
-                var current = _dictionaryValues[ strToParse[ i ].ToString() ];
-                count++;
-                if (count < len)
+                var isExist = dictionaryValues.ContainsKey(strToParse[ i ].ToString());
+                if (isExist)
                 {
-                    var next = _dictionaryValues[ strToParse[ i + 1 ].ToString() ];
-                    if (current < next)
+                    var current = dictionaryValues[ strToParse[ i ].ToString() ];
+                    count++;
+                    if (count < len)
                     {
-                        resultArabicNumber -= current;
+                        var next = dictionaryValues[ strToParse[ i + 1 ].ToString() ];
+                        if (current < next)
+                        {
+                            resultArabicNumber -= current;
+                        }
+                        else
+                        {
+                            resultArabicNumber += current;
+                        }
                     }
                     else
                     {
                         resultArabicNumber += current;
                     }
+
                 }
                 else
                 {
-                    resultArabicNumber += current;
+                    throw new ArgumentException("There is no such key");
```

Comment:
please do a clear message here. User won't understand this.
Something like: Failed to parse value 'X' as roman, symbol 'Y' is not a valid roman literal.